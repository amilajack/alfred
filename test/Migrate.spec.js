// @flow
import path from 'path';
import espree from 'espree';
import { expect as chaiExpect } from 'chai';
import Providers, { handleInput } from '../src/providers';

jest.setTimeout(20000);

function parseCodeSnippets(codeSnippets: Array<string> | void) {
  if (codeSnippets) {
    codeSnippets.map(code =>
      espree.parse(code, {
        sourceType: 'module'
      })
    );
  }
}

describe.skip('Migrate', () => {
  const defaultConfig = {
    packageJsonPath: __dirname,
    write: false,
    unsafe: false,
    verbose: false
  };

  it('should migrate basic file', async () => {
    const tmpFile = path.join(__dirname, 'fixtures', 'basic-test.ts');
    const result = await Providers({
      files: [tmpFile],
      ...defaultConfig
    });
    expect(result).toMatchSnapshot();
    parseCodeSnippets(result);
  });

  it('should migrate basic file with EslintProvider', async () => {
    const tmpFile = path.join(__dirname, 'fixtures', 'comma-test.ts');
    const result = await Providers({
      files: [tmpFile],
      ...defaultConfig
    });
    expect(result).toMatchSnapshot();
    parseCodeSnippets(result);
  });

  it('should perform unsafe transformations', async () => {
    const result1 = await Providers({
      ...defaultConfig,
      unsafe: true,
      files: [
        path.join(__dirname, 'fixtures', 'unsafe-transformation-test.ts'),
        path.join(__dirname, 'fixtures', 'class-test.ts')
      ]
    });
    expect(result1).toMatchSnapshot();
    parseCodeSnippets(result1);

    const result2 = await Providers({
      ...defaultConfig,
      unsafe: true,
      files: [
        path.join(__dirname, 'fixtures', 'unsafe-transformation-test.ts'),
        path.join(__dirname, 'fixtures', 'class-test.ts')
      ]
    });
    expect(result2).toMatchSnapshot();
    parseCodeSnippets(result2);
  });

  it('should fail on non-existent files', async () => {
    try {
      await Providers({
        ...defaultConfig,
        files: [
          path.join(__dirname, 'fixtures', 'foo-test.ts'),
          path.join(__dirname, 'fixtures', 'class-test.ts')
        ]
      });
    } catch (e) {
      expect({ ...e, path: '' }).toMatchSnapshot();
    }
  });

  it('should run against directories', async () => {
    const result = await Providers({
      ...defaultConfig,
      files: [
        path.join(__dirname, 'fixtures', 'express-hello-world', 'foobar.js'),
        path.join(__dirname, 'fixtures', 'express-hello-world', 'index.js')
      ]
    });
    expect(result).toMatchSnapshot();
    parseCodeSnippets(result);
  });

  it.skip("should not parse .gitignore'd files", async () => {
    const files = await handleInput({
      files: ['.'],
      packageJsonPath: path.join(__dirname, '..', 'package.json'),
      ...defaultConfig
    });

    console.log(files);

    files.forEach(file => {
      chaiExpect(file).to.not.include('node_modules');
    });
  });

  it.skip('should allow selection of manual tests', () => {});

  it.skip('should not write files if one provider fails', async () => {});

  it.skip('should fail on non-js extensions', async () => {});

  it.skip('should fail on parsing coffeescript', async () => {});
});
