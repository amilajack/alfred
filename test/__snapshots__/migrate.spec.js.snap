// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Migrate should migrate basic file 1`] = `
Array [
  "import fs from 'fs';
import path from 'path';


const foo = 'asdfa';
console.log(foo);
",
]
`;

exports[`Migrate should migrate basic file with EslintProvider 1`] = `
Array [
  "const some = 'foo';
const bar = 'baz';
",
]
`;

exports[`Migrate should perform unsafe transformations 1`] = `
Array [
  "import foo from 'bar';
import cow from 'boo';
",
]
`;
